### PR TITLE
rfb: add RFB parser

### PR DESCRIPTION
**Pull request by satta**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [#3501](https://redmine.openinfosecfoundation.org/issues/3501)

Describe changes:
- Adds a Rust parser for the RFB (Remote Framebuffer, VNC) protocol up to the end of the client-server handshake, producing metadata for logging and two keywords (`rfb.name`, `rfb.sectype`) for signature writing. Please see the included documentation (EVE-JSON fields, keywords) for more information.

### CODE REVIEW

File: suricata.yaml.in [hunk 1]
```diff
@@ -264,6 +264,7 @@ outputs:
         - ikev2
         - krb5
         - snmp
+        #- rfb
```
Comment (victorjulien) [inline suricata.yaml.in]: I'd say we enable this by default, unless there are reasons this shouldn't be done.

File: suricata.yaml.in [hunk 2]
```diff
@@ -713,6 +714,8 @@ pcap-file:
 # "detection-only" enables protocol detection only (parser disabled).
 app-layer:
   protocols:
+    rfb:
+      enabled: no
```
Comment (victorjulien) [inline suricata.yaml.in]: same here, lets enable by default in our dev phase

File: src/output-json-rfb.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/* Copyright (C) 2018 Open Information Security Foundation
```
Comment (victorjulien) [inline src/output-json-rfb.h]: 2020

File: src/output-json-rfb.c [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+void JsonRFBLogRegister(void)
+{
+    /* Register as an eve sub-module. */
+    OutputRegisterTxSubModule(LOGGER_JSON_RFB, "eve-log",
+        "JsonRFBLog", "eve-log.rfb",
+        OutputRFBLogInitSub, ALPROTO_RFB, JsonRFBLogger,
+        JsonRFBLogThreadInit, JsonRFBLogThreadDeinit, NULL);
+
+    SCLogNotice("RFB JSON logger registered.");
```
Comment (victorjulien) [inline src/output-json-rfb.c]: left over debug statement

File: src/output-json-rfb.c [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    OutputCtx *output_ctx = SCCalloc(1, sizeof(*output_ctx));
+    if (unlikely(output_ctx == NULL)) {
+        SCFree(rfblog_ctx);
+        return result;
+    }
+    output_ctx->data = rfblog_ctx;
+    output_ctx->DeInit = OutputRFBLogDeInitCtxSub;
+
+    SCLogNotice("RFB log sub-module initialized.");
```
Comment (victorjulien) [inline src/output-json-rfb.c]: debug

File: src/output-json-rfb.c [hunk 6]
```diff
@@ -0,0 +1,174 @@
+/* Copyright (C) 2018 Open Information Security Foundation
```
Comment (victorjulien) [inline src/output-json-rfb.c]: 2020

File: src/detect-rfb-name.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
+/* Copyright (C) 2019 Open Information Security Foundation
```
Comment (victorjulien) [inline src/detect-rfb-name.h]: 2020

File: src/detect-rfb-name.c [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+#include "detect-rfb-name.h"
+#include "stream-tcp.h"
+
+#include "rust.h"
+#include "app-layer-rfb.h"
+#include "rust-bindings.h"
+
+#define KEYWORD_NAME "rfb.name"
+#define KEYWORD_DOC  ""
```
Comment (victorjulien) [inline src/detect-rfb-name.c]: unused?

File: src/detect-rfb-name.c [hunk 9]
```diff
@@ -0,0 +1,120 @@
+/* Copyright (C) 2019 Open Information Security Foundation
```
Comment (victorjulien) [inline src/detect-rfb-name.c]: 2020

File: rust/src/rfb/mod.rs [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/* Copyright (C) 2018 Open Information Security Foundation
```
Comment (victorjulien) [inline rust/src/rfb/mod.rs]: 2020

**Comment by victorjulien** [comment]
@jlucovsky can you review the C parts?
@jasonish can you review the Rust parts?

**Comment by satta** [comment]
New PR: #4601

File: src/output-json-rfb.c [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+ *
+ * \author Frank Honza <frank.honza@dcso.de>
+ *
+ * Implement JSON/eve logging app-layer RFB.
+ */
+
+#include "suricata-common.h"
+#include "debug.h"
+#include "detect.h"
+#include "pkt-var.h"
```
Comment (jlucovsky) [inline src/output-json-rfb.c]: needed?

**Comment by satta** [comment]
Thanks @jlucovsky! We'll address it in v2.

**Comment by satta** [comment]
New PR: #4610

**Pull request closed without merging**